WooCommerce/Classes/ViewRelated/Dashboard/Settings/Help/ApplicationLogDetailViewController.swift

```diff
@@ -0,0 +1,86 @@
+import UIKit
+
+class ApplicationLogDetailViewController: UIViewController {
+
+    /// The contents of the log file.
+    ///
+    var logText: String
+
+    /// The log file's creation date as a formatted string.
+    ///
+    var logDate: String
+
+    /// The textview for displaying the log data
+    ///
+    @IBOutlet private var textview: UITextView!
+
+
+    // MARK: - Overridden Methods
+    //
+    
+    /// Designated initializer
+    ///
+    init(with contents: String, for date: String) {
+        self.logText = contents
+        self.logDate = date
+
+        super.init(nibName: "ApplicationLogDetailViewController", bundle: Bundle.main)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        self.logText = ""
+        self.logDate = ""
+        super.init(coder: aDecoder)
+    }
+
```

Comment:
Initializers on a view controller isn't something I'm used to creating. Are these correct?